# 2020 Winter: BIODS 215

Welcome to BIODS215 Topics in Biomedical Data Science: Large-scale inference!

![Course flyer for BIODS 215, Winter 2020]({{ site.url }}/lecture_material/Introduction/2020/BDS215-flyer_2020.png)

- [Website for 2018 Winter quarter is available here](2018.md)
- [Website for 2017 Spring quarter is available here](2017.md)
- Go back to [the top page]({{ site.url }})

## Teaching team

We highly encourage everyone to use Piazza to contact with the teaching team.
However, you can also reach out to us using email.
The mailing list to the entire teaching team is: `biods215-2020 at lists.stanford.edu`.

### Instructors

- [Manuel A. Rivas](http://rivaslab.stanford.edu) (email: `mrivas at stanford.edu`), Office hours: Tuesdays 4:20 - 5:20 (MSOB x321)
- [James Zou](https://www.james-zou.com/) (email: `jamesz at stanford.edu`)

#### TA

- Yosuke Tanigawa (email: `ytanigaw at stanford.edu`), Office hours: Tuesdays 11:00 - 12:00 (MSOB x391)

## Links

- [Canvas](https://canvas.stanford.edu/courses/113896)
- [Piazza](http://piazza.com/stanford/winter2020/biods215)
- [Gradescope](https://www.gradescope.com/courses/77264), Entry Code:MZGP3Y

## Course content plan

| #  | Date      | Lecturer      | Room     | Topic                                                                     |
|----|-----------|---------------|----------|---------------------------------------------------------------------------|
| 1  | 1/7/2020  | Manuel        | MSOBx303 | Overview of emerging topics in biomedical data science                    |
| 2  | 1/9/2020  | Manuel        | MSOBx303 | Topics in Linear Algebra                                                  |
| 3  | 1/14/2020 | Manuel        | MSOBx303 | Optimization                                                              |
| 4  | 1/16/2020 | Manuel        | MSOBx303 | Mixture Models                                                            |
| 5  | 1/21/2020 | James         | MSOBx303 | Machine Learning for CRISPR editing                                       |
| 6  | 1/23/2020 | Guest lecture | MSOBx303 | Ismael Lemhadri: "lassoNet"                                               |
| 7  | 1/28/2020 | Manuel        | MSOBx303 | Causal inference using instrumental variables                             |
| 8  | 1/30/2020 | Manuel        | MSOBx303 | Causal inference using instrumental variables II: Mendelian randomization |
| 9  | 2/4/2020  | Manuel        | MSOBx303 | Gaussian Process regression                                               |
| 10 | 2/6/2020  | Yosuke        | MSOBx303 | Reproducible large-scale inference                                        |
| 11 | 2/11/2020 | James         | MSOBx303 | Deep Learning I                                                           |
| 12 | 2/13/2020 | James         | *LK120*  | Deep Learning II                                                          |
| 13 | 2/18/2020 | Manuel        | MSOBx303 | Risk Models                                                               |
| 14 | 2/20/2020 | Manuel        | MSOBx303 | Survival risk models                                                      |
| 15 | 2/25/2020 | Manuel        | MSOBx303 | Multitask risk modeling                                                   |
| 16 | 2/27/2020 | Manuel       | *MSOBx393* | False discovery rates                                                    |
| 17 | 3/3/2020  | Final project | MSOBx303 | Final project presentation                                                |
| 18 | 3/5/2020  | Final project | MSOBx303 | Final project presentation                                                |
| 19 | 3/10/2020 | James         | MSOBx303 | Zou Lab research presentation                                             |
| 20 | 3/12/2020 | Manuel        | MSOBx303 | Rivas Lab research presentation                                           |

## Assignments

Please submit all the assignments via Gradescope.

Late day policy. Students have 6 late days in total. We allow a maximum of 2 days per assignment.

### Reading Materials

We ask students to write a paragraph about the reading materials and/or the corresponding lecture. Here is the instructions:

- Please read the reading materials posted on the class website and write a paragraph about it. You may write more if you want.
- You can write anything regarding the reading material and/or the corresponding lecture.
- You may also include comments and/or request regarding the teaching style. We appreciate your feedback!
- Please submit your answer through Gradescope as one pdf document.

### Problem set 1

- Release date: 1/14/2020, Due date: 1/30/2020 (extended from 1/28/2020)
- [Problem set 1](https://canvas.stanford.edu/courses/113896/files?preview=5401649)
- [Answer key](https://canvas.stanford.edu/courses/113896/files?preview=5581673)
- Regrade request due: 2/25 (Tue.) 11:59 pm.

### Problem set 2

- Release date: 1/30/2020, Due date: 2/18/2020
- [Problem set 2](https://canvas.stanford.edu/courses/113896/files?preview=5476832)
- Additional reference
  - [Hartwig, F. P., Davey Smith, G. & Bowden, J. Robust inference in summary data Mendelian randomization via the zero modal pleiotropy assumption. Int J Epidemiol 46, 1985–1998 (2017).](https://doi.org/10.1093/ije/dyx102)
- [Answer key](https://canvas.stanford.edu/courses/113896/files?preview=5697014)
- [Sample code](https://canvas.stanford.edu/courses/113896/files?preview=5697006)
- Regrade request due: 3/17 (Tue.) 11:59 pm.

### Class project

We think the class project is a great opportunity for you to use some of the methods you will learn in the class. To allocate sufficient time to work on the project, we would like to have the brief project proposal by the third week of the quarter, 1/23/2020.

#### [Project proposal]({{ site.url }}/projects/2020/BIODS215-2020_Class_projects.pdf)

- Release date: 1/9/2020, Due date: 1/23/2020

Please check [this document]({{ site.url }}/projects/2020/BIODS215-2020_Class_projects.pdf) for more details.

#### Project write-up

- Due date: 3/12/2020
- There is no page limit, but We would like to see the following contents:
  - Research question
  - Motivation/Background
  - Method
    - Please clearly describe the statistical/deep learning model you've used in the class project
  - Results & Discussion
    - Make sure to include all the main findings from the analysis.
    - For figures and tables, please use clear labels and provide legend so that we have a clear ideas what is presented on the figures/tables.
- For group project, please clarify the contribution of each group member in the final project write-up.
- If you are working in a lab, you may use some of the results from your research as the class project, given that PI(s) are aware and supportive about it. If you were to take this route, we will ask you to clarify the relevance of the project to the materials covered in the class.

## Lecture materials

We will post the list of lecture slides and reading materials here.

### Lecture 1. Overview of emerging topics in biomedical data science

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5364766)
- [D. Donoho. 50 years of Data Science]({{ site.url }}/readings/50YearsDataScience_Donoho2015.pdf)
- [Chapter 18 and Epilogue of Computer Age Statistical Inference](https://web.stanford.edu/~hastie/CASI_files/PDF/casi.pdf)

### Lecture 2. Topics in Linear Algebra

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5391237)
- [Tibshirani, R. In praise of sparsity and convexity. in Past, Present, and Future of Statistical Science (eds. Lin, X. et al.) 497–505 (Chapman and Hall/CRC, 2014). doi:10.1201/b16720-47.](http://statweb.stanford.edu/~tibs/ftp/tibs-copss.pdf)

### Lecture 3. Optimization

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5392745)
- [Constrained Optimization — Computational Statistics and Statistical Computing 1.0 documentation.](http://people.duke.edu/~ccc14/sta-663-2018/notebooks/S09H_Constrained_Optimization.html)

### Lecture 4. Mixture Models

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5406334)
- [Rivas, M. A. et al. Effect of predicted protein-truncating genetic variants on the human transcriptome. Science 348, 666–669 (2015).](https://doi.org/10.1126/science.1261877)
- [Rivas, M. A. et al. Insights into the genetic epidemiology of Crohn’s and rare diseases in the Ashkenazi Jewish population. PLOS Genetics 14, e1007329 (2018).](https://doi.org/10.1371/journal.pgen.1007329)
- [Flynn, E. et al. Sex-specific genetic effects across biomarkers. bioRxiv 837021 (2019) doi:10.1101/837021.](https://doi.org/10.1101/837021)

### Lecture 5. Machine Learning for CRISPR editing

- [Zou, J. et al. A primer on deep learning in genomics. Nature Genetics (2018).](https://doi.org/10.1038/s41588-018-0295-5)

### Lecture 6. LassoNet (guest lecture)

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5441869)
- [Lemhadri, I., Ruan, F. & Tibshirani, R. A neural network with feature sparsity. arXiv:1907.12207 [cs, stat] (2019).](https://arxiv.org/abs/1907.12207)

### Lecture 7. Causal inference using instrumental variables

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5464868)
- [Burgess, S., Foley, C. N. & Zuber, V. Inferring Causal Relationships Between Risk Factors and Outcomes from Genome-Wide Association Study Data. Annual Review of Genomics and Human Genetics 19, 303–327 (2018).](https://doi.org/10.1097/01.ede.0000222409.00878.37) (please clink "View on Journal Site")
- [Hernan, M. A. & Robins, J. M. Instruments for Causal Inference: An Epidemiologist??s Dream? Epidemiology 17, 360–372 (2006).](https://doi.org/10.1146/annurev-genom-083117-021731)
- [P Cannon, C. IMPROVE-IT Trial: A Comparison of Ezetimibe/Simvastatin versus Simvastatin Monotherapy on Cardiovascular Outcomes After Acute Coronary Syndromes. in (2014).](http://my.americanheart.org/idc/groups/ahamah-public/@wcm/@sop/@scon/documents/downloadable/ucm_469598.pdf)
- [Burgess, S., Foley, C. N. & Zuber, V. Inferring Causal Relationships Between Risk Factors and Outcomes from Genome-Wide Association Study Data. Annual Review of Genomics and Human Genetics 19, 303–327 (2018).](https://doi.org/10.1016/j.jacc.2015.02.020)

### Lecture 8. Causal inference using instrumental variables II: Mendelian randomization

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5474458)
- [Darrous, L., Mounier, N. & Kutalik, Z. Simultaneous estimation of bi-directional causal effects and heritable confounding from GWAS summary statistics. medRxiv 2020.01.27.20018929 (2020) doi:10.1101/2020.01.27.20018929.](https://doi.org/10.1101/2020.01.27.20018929 )
- [Burgess, S., Foley, C. N. & Zuber, V. Inferring Causal Relationships Between Risk Factors and Outcomes from Genome-Wide Association Study Data. Annual Review of Genomics and Human Genetics 19, 303–327 (2018).](https://doi.org/10.1146/annurev-genom-083117-021731)

### Lecture 9. Gaussian Process regression

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5502031)
- [Chapter 2 Gaussian Process regression, Rasmussen, C. E. & Williams, C. K. I. Gaussian processes for machine learning. (MIT Press, 2006).](http://www.gaussianprocess.org/gpml/chapters/RW2.pdf)
- [1.7. Gaussian Processes — scikit-learn 0.22.1 documentation.](https://scikit-learn.org/stable/modules/gaussian_process.html)
- [Bayesian non-parametrics with Gaussian Processes]({{ site.url }}/lecture_material/GaussianProcess/2017/GP_Practice.pdf)

### Lecture 10. Reproducible large-scale inference

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5523829)
- [Ioannidis, J. P. A. . Why Most Published Research Findings Are False. PLoS Medicine 2, e124 (2005).](https://doi.org/10.1371/journal.pmed.0020124)
- [Write Your Own R Packages.](https://www.pixelstech.net/article/1571487185-Write-Your-Own-R-Packages)
- [Coding habits for data scientists. ThoughtWorks (2019).](https://www.thoughtworks.com/insights/blog/coding-habits-data-scientists)
- [Thomas, K. et al. Jupyter Notebooks - a publishing format for reproducible computational workflows. Stand Alone 87–90 (2016).](https://doi.org/10.3233/978-1-61499-649-1-87)

### Lecture 11. Deep Learning I

- [Interactive tutorial to build a convolutional neural network to discover DNA-binding motifs in Zou, J. et al. A primer on deep learning in genomics. Nature Genetics (2018)](https://colab.research.google.com/drive/17E4h5aAOioh5DiTo7MZg4hpL6Z_0FyWr).

### Lecture 12. Deep Learning II

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5572955)
- [Zou, J. et al. A primer on deep learning in genomics. Nature Genetics (2018).](https://doi.org/10.1038/s41588-018-0295-5)

### Lecture 13. Risk Models

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5580225)
- [Jostins, L. & Barrett, J. C. Genetic risk prediction in complex disease. Hum Mol Genet 20, R182–R188 (2011).](https://doi.org/10.1093/hmg/ddr378)
- [Qian, J. et al. A Fast and Flexible Algorithm for Solving the Lasso in Large-scale and Ultrahigh-dimensional Problems. bioRxiv 630079 (2019).](https://doi.org/10.1101/630079)

### Lecture 14. Survival risk models

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5591524)
- [Bewick, V., Cheek, L. & Ball, J. Statistics review 12: Survival analysis. Crit Care 8, 389 (2004).](https://doi.org/10.1186/cc2955)
- [Li, R. et al. Fast Lasso method for Large-scale and Ultrahigh-dimensional Cox Model with applications to UK Biobank. bioRxiv 2020.01.20.913194 (2020).](https://doi.org/10.1101/2020.01.20.913194)

### Lecture 15. Multitask risk modeling

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5623946)
- [Tanigawa, Y.\*, Li, J.\* et al. Components of genetic associations across 2,138 phenotypes in the UK Biobank highlight adipocyte biology. Nat Commun 10, 1–14 (2019).](https://doi.org/10.1038/s41467-019-11953-9)
- [Aguirre, M. et al. Polygenic risk modeling with latent trait-related genetic components. bioRxiv 808675 (2019) doi:10.1101/808675.](https://doi.org/10.1101/808675)

### Lecture 16. False discovery rates

- [Lecture slides (on Canvas)](https://canvas.stanford.edu/courses/113896/files?preview=5633013)
- [Stephens, M. False discovery rates: a new deal. Biostatistics 18, 275–294 (2017).](https://doi.org/10.1093/biostatistics/kxw041)

### Lecture 17 & 18. Final project presentation

### Lecture 19. Zou Lab research presentation

- [James Zou's research group](https://www.james-zou.com/)
- [Ouyang, D. et al. Interpretable AI for beat-to-beat cardiac function assessment. medRxiv 19012419 (2019).](https://doi.org/10.1101/19012419)
- [Ghorbani, A. & Zou, J. Neuron Shapley: Discovering the Responsible Neurons. arXiv:2002.09815 [cs, stat] (2020).](https://arxiv.org/abs/2002.09815)
- [Abid, A. et al. Gradio: Hassle-Free Sharing and Testing of ML Models in the Wild. arXiv:1906.02569 [cs, stat] (2019).](https://arxiv.org/abs/1906.02569)

### Lecture 20. Rivas Lab research presentation

- [Manuel A. Rivas's research group](http://rivaslab.stanford.edu)
